readthedocs/projects/tasks/utils.py

```diff
@@ -271,6 +271,14 @@ def deprecated_config_file_used_notification():
     queryset = User.objects.filter(username__in=users, profile__banned=False).order_by(
         "id"
     )
+
+    # Exclude users that already have an unread notification.
+    # Our notifications backend already performs de-duplication of notifications automatically.
+    # However, this is only to speed up things here since this loop takes the most time.
+    queryset = queryset.exclude(
+        message__message__startswith="Your project(s)", message__read=False
+    )
```

Comment:
This idea is good for onsite notifications. However, it does not work well for email notifications. So, this code may not be 100% accurate and I may need to re-think it.